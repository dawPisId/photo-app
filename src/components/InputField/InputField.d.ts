export interface InputFieldProps {
  /**
   * Input label displayed on the top left of the box.
   */
  label: string;
  /**
   * Value displayed in the textbox.
   */
  fillValue: string;
}
