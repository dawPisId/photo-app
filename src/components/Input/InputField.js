import { InputBox, InputLabel, InputWrapper } from "./InputField.styled";

import React from "react";
import SearchIcon from "@material-ui/icons/Search";

const InputField = ({label, action}) => (
  <InputWrapper>
    <SearchIcon />
    <InputLabel>{label}</InputLabel>
    <InputBox type="text" onChange = {action}/>
  </InputWrapper>
);

export default InputField;
