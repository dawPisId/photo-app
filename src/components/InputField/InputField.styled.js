import { Colors } from "../../Colors";
import styled from "styled-components";

export const InputWrapper = styled.div`
  display: inline-flex;
  background-color: ${Colors.transparent};
  position: relative;
  align-items: center;
  margin: 0rem 0.5rem;
  padding: 0px;
  border: 1px solid lightgrey;
  border-radius: 5px;
  svg {
    vertical-align: center;
    color: grey;
    margin-left: 0.5rem;
  }
`;
export const InputBox = styled.input`
  font-family: "Roboto", sans-serif;
  background-color: ${Colors.transparent};
  font-size: 1.25rem;
  margin: 0.4rem 0rem;
  padding-left:1rem;
  width:100%;
  border-width: 0px;
  &:focus {
    outline: none;
  }
  @media(max-width:640px){
    font-size:1rem;
  }
  @media(max-width:530px){
    font-size:0.75rem;
  }
`;
export const InputLabel = styled.span`
  position: absolute;
  color: grey;
  background-color:white;
  top: -0.6rem;
  left: 0.6rem;
  font-size: 0.75rem;
`;
