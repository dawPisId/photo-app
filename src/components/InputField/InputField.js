import { InputBox, InputLabel, InputWrapper } from "./InputField.styled";

import React from "react";

const InputField = ({ label, fillValue }) => (
  <InputWrapper>
    <InputLabel>{label}</InputLabel>
    <InputBox type="text" value={fillValue} readOnly={true} />
  </InputWrapper>
);

export default InputField;
