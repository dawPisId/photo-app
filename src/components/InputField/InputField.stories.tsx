import { ComponentMeta, ComponentStory } from "@storybook/react";

import InputField from "./InputField";
import React from "react";

export default {
  title: "Photo App/InputField",
  component: InputField,
} as ComponentMeta<typeof InputField>;

const Template: ComponentStory<typeof InputField> = (args) => (
  <InputField {...args} />
);

export const Main = Template.bind({});
Main.args = {
  label: "one",
  fillValue: "two",
};
