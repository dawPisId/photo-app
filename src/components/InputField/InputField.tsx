import { InputBox, InputLabel, InputWrapper } from "./InputField.styled";

import { InputFieldProps } from "./InputField.d";
import React from "react";

const InputField: React.FC<InputFieldProps> = ({ label, fillValue }) => (
  <InputWrapper>
    <InputLabel>{label}</InputLabel>
    <InputBox type="text" value={fillValue} readOnly={true} />
  </InputWrapper>
);

export default InputField;
