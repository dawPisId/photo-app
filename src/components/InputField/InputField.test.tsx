import InputField from "./InputField";
import React from "react";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(<InputField label="one" fillValue="two" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
