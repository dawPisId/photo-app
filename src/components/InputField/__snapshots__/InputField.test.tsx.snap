// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdnxRM kxhzw"
>
  <span
    className="sc-dlnjwi joyBPH"
  >
    one
  </span>
  <input
    className="sc-gtsrHT gweHHT"
    readOnly={true}
    type="text"
    value="two"
  />
</div>
`;
